File: src/MenuScene.cc
#include "MenuScene.h"

#include <cheerp/client.h>
#include <cheerp/clientlib.h>

#include "GlobalFactories.h"

using namespace client;
using namespace cheerp;


MenuScene::MenuScene()
{
  HTMLElement * body = document.get_body();

  //title
  title = document.createElement("h1");
  title->set_textContent("Detention!");

  //subtitle
  subtitle = document.createElement("h2");
  subtitle->set_textContent("the Family Favourite Corporal Punishment Game");

  //picture
  button = static_cast<HTMLImageElement *>(document.createElement("img"));
  button->set_src("assets/begin.png");
  button->set_alt("press it or die");
  button->set_onclick(Callback([this]()
  {
    countDown = 5;
    sound->play();
  }));

  //bottom text
  bottom = document.createElement("div");

  //add them in
  body->appendChild(title);
  body->appendChild(subtitle);
  body->appendChild(button);
  body->appendChild(bottom);

  //load the breaking glass sound
  sound = static_cast<HTMLAudioElement *>(document.createElement("audio"));
  sound->set_src(new String("assets/se/smash.wav"));
}

MenuScene::~MenuScene()
{
  //remove the html elements from the document
  HTMLElement * body = document.get_body();
  body->removeChild(title);
  body->removeChild(subtitle);
  body->removeChild(button);
  body->removeChild(bottom);
}

Scene * MenuScene::update(float deltaTime)
{
  if (countDown > 0)
  {
    countDown -= deltaTime;

    //add another fuck yeah!
    HTMLElement * fuckYeah = document.createElement("h2");
    fuckYeah->set_textContent("fuck yeah!");
    bottom->appendChild(fuckYeah);
    window.scrollTo(0,document.get_body()->get_scrollHeight());

    //if it's time to go
    if (countDown <= 0)
    {
      return GlobalFactories::cutSceneFactory.make(new String("ebola"));
    }
  }
  return this;
}

File: src/MenuScene.js
//a menu where you start the game
//7/6/2015

$include SoundPlayer.js$
$include CutScene.js$
$include assets.js$

const FIRST_CUT_SCENE = "1.pig";


//creates the MenuScene with no arguments
var MenuScene = function()
{
  //set some things
  this.countdown = -1.0;

  //create the heading
  this.heading = document.createElement("h1");
  this.heading.innerHTML = "Detention!";
  document.body.appendChild(this.heading);

  //create the subtitle
  this.subTitle = document.createElement("h2");
  this.subTitle.innerHTML = "the Family Favourite Corporal Punishment Game";
  document.body.appendChild(this.subTitle);

  //create the picture
  this.picture = new Image();
  this.picture.src = "assets/graphic/begin.png";
  this.picture.noise = assets.sEs["nice.wav"];
  this.picture.owner = this;
  this.picture.onclick = function()
  {
    this.owner.countdown = 2.0;
    this.noise.play();
  };
  document.body.appendChild(this.picture);

  //create a div to put the 'fuck yeah's in
  this.fuckYeahs = document.createElement("div");
  document.body.appendChild(this.fuckYeahs);
};


//destroys the contents of the MenuScene with no arguments
MenuScene.prototype.delete = function()
{
  document.body.removeChild(this.heading);
  document.body.removeChild(this.subTitle);
  document.body.removeChild(this.picture);
  document.body.removeChild(this.fuckYeahs);
}


//updates the MenuScene, although it does nothing
//deltaTime is the time since last time
//returns whichever scene is to continue into the future
MenuScene.prototype.update = function(deltaTime)
{
  if (this.countdown > 0)
  {
    var fuckYeah = document.createElement("h1");
    fuckYeah.innerHTML = "fuck yeah";
    this.fuckYeahs.appendChild(fuckYeah);
    window.scrollTo(0,document.body.scrollHeight);

    this.countdown -= deltaTime;
    if (this.countdown <= 0)
    {
      return makeCutScene(assets.cutSceneDatas[FIRST_CUT_SCENE].getData());
    }
  }
  return this;
};
